File: 디자인패턴/java 디자인 패턴/Observer/OddBettingPlayer.java
public class OddBettingPlayer extends Player {
    protected OddBettingPlayer(String name) {
        super(name);
    }

    @Override
    protected void ProcessDiceBetting(int Dice_Num) {
        System.out.println("\nPlayer: " + getPlayer_name());
        if (Dice_Num%2==1) System.out.println("Win!");
        else System.out.println("Lose!");
    }  
}

File: 디자인패턴/Python 디자인 패턴/Observer/OddBettingPlayer.py
from Player import*
class OddBettingPlayer(Player):
    def __init__(this, name: str) -> None:
        super().__init__(name)
    
    def Update(this, diceNum: int) -> None:
        print(f"\nPlayer: {this.getPlayerName()}")
        if(diceNum%2):print("Win")
        else:print("Lose!")
        pass
    pass